src/firefly/java/edu/caltech/ipac/firefly/server/vispush/PushJob.java

```diff
@@ -152,9 +152,20 @@ public static boolean pushRemoveMask(String maskId) {
         return true;
     }
 
-    public static boolean isBrowserClientActive(String ipString) {
-        String channel= ServerContext.getRequestOwner().getEventChannel();
-        return ServerEventManager.getActiveQueueChannelCnt(channel)>1;
+    public static int getBrowserClientActiveCount(String channel, int tryTime) {
+        if (channel==null) channel= ServerContext.getRequestOwner().getEventChannel();
+        int cnt=  ServerEventManager.getActiveQueueChannelCnt(channel);
+        long endTry= System.currentTimeMillis()+tryTime;
+
+        try {
+            while (tryTime>0 && cnt==0 && System.currentTimeMillis()<endTry) {
+                Thread.sleep(200);
```

Comment:
No, it is right.  I guess I do need to document it more.  `tryTime` is the about of time that is will check for.  It polls every 200 ms until success or tryTime runs out.  This way it will return quite fast.  `tryTime` is really the time until it gives up. The polling is only looking at an array that is updated in another thread.